code/modules/mob/living/carbon/xenomorph/castes/ravager/abilities_ravager.dm

```diff
@@ -233,3 +233,156 @@
+	"<span class='highdanger'>We bellow as our fury overtakes us! RIP AND TEAR!</span>")
+	X.do_jitter_animation(1000)
+
+
+	//Roar SFX; volume scales with rage
+	playsound(X.loc, 'sound/voice/alien_roar2.ogg', clamp(100 * rage_power, 25, 80), 0)
+
+	var/bonus_duration
+	if(rage_power > 0.5) //If we're super pissed it's time to get crazy
+		var/datum/action/xeno_action/charge = X.actions_by_path[/datum/action/xeno_action/activable/charge]
+		var/datum/action/xeno_action/ravage = X.actions_by_path[/datum/action/xeno_action/activable/ravage]
+		if(charge)
+			charge.clear_cooldown() //Reset charge cooldown
+		if(ravage)
+			ravage.clear_cooldown() //Reset ravage cooldown
+
+		for(var/mob/living/witness in hearers(rage_power_radius, X)) //Roar that applies cool SFX and some soft CC
+
+			if(!witness.hud_used)
+				continue
+
+			var/obj/screen/plane_master/floor/OT = witness.hud_used.plane_masters["[FLOOR_PLANE]"]
+			var/obj/screen/plane_master/game_world/GW = witness.hud_used.plane_masters["[GAME_PLANE]"]
+
+			addtimer(CALLBACK(OT, /atom.proc/remove_filter, "rage_outcry"), 1 SECONDS)
+			GW.add_filter("rage_outcry", 2, list("type" = "radial_blur", "size" = 0.07))
+			animate(GW.get_filter("rage_outcry"), size = 0.12, time = 5, loop = -1)
+			OT.add_filter("rage_outcry", 2, list("type" = "radial_blur", "size" = 0.07))
+			animate(OT.get_filter("rage_outcry"), size = 0.12, time = 5, loop = -1)
+			addtimer(CALLBACK(GW, /atom.proc/remove_filter, "rage_outcry"), 1 SECONDS)
+
+	for(var/turf/affected_tiles as() in RANGE_TURFS(rage_power_radius, X.loc))
+		affected_tiles.Shake(4, 4, 1 SECONDS) //SFX
+
+	for(var/mob/living/L in hearers(rage_power_radius, X)) //Roar that applies cool SFX and some soft CC
+		shake_camera(L, 1 SECONDS, 1)
+		L.Shake(4, 4, 1 SECONDS) //SFX
+
+		if(L.stat == DEAD) //We don't care about the dead
+			continue
+
+		if(isxeno(L))
+			var/mob/living/carbon/xenomorph/friendly_check = L
+			if(friendly_check.issamexenohive(X)) //No friendly fire
+				continue
+
+	X.add_filter("ravager_rage_outline", 5, outline_filter(1.5, COLOR_RED)) //Set our cool aura; also confirmation we have the buff
+
+
+	rage_plasma = min(X.xeno_caste.plasma_max - X.plasma_stored, X.xeno_caste.plasma_max * rage_power) //Calculate the plasma to restore (and take away later)
+	X.plasma_stored += rage_plasma //Regain a % of our maximum plasma scaling with rage
+
+	rage_sunder = min(X.sunder, rage_power * 100) //Set our temporary Sunder recovery
+	X.adjust_sunder(-1 * rage_sunder) //Restores up to 100 Sunder temporarily.
+
+	X.xeno_melee_damage_modifier += rage_power  //Set rage melee damage bonus
+
+	X.add_movespeed_modifier(MOVESPEED_ID_RAVAGER_RAGE, TRUE, 0, NONE, TRUE, X.xeno_caste.speed * 0.5 * rage_power) //Set rage speed bonus
+
+	//Too angry to be stunned/slowed/staggered/knocked down
+	ADD_TRAIT(X, TRAIT_STUNIMMUNE, RAGE_TRAIT)
+	ADD_TRAIT(X, TRAIT_SLOWDOWNIMMUNE, RAGE_TRAIT)
+	ADD_TRAIT(X, TRAIT_STAGGERIMMUNE, RAGE_TRAIT)
```

Comment:
he, don't like this as well.

Reply:
Not really impactful due to the immunities granted by Endure tbh, and you probably want to pair it with Endure anyways. I don't feel too strongly about this.

Reply:
Ravager needs it , that or stun immunity all togheter , its kinda silly for someone who is meant to come to marines and rip + tear  them would get absolutely destroyed by stagger or knockdown from snipers / shotguns